packages/backend/src/peripherals/database/shared/Database.ts

```diff
@@ -35,6 +52,9 @@ export class Database {
           path.join(__dirname, '..', 'migrations'),
         ),
       },
+      pool: {
+        afterCreate: callbackify(this.ensureServerVersion.bind(this)),
+      },
```

Comment:
Do we need to check it after creating every connection? Doesn't really look optimal for me. Is there a way to do it only on startup of the app maybe?

Reply:
As far as my understanding goes it does not happen that often - there's only a handful of connections to postgres server and they are pooled and reused by node process. But in fact just checking it in `Application.ts` will simplify the code and will utilise simpler concepts.